flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -184,8 +185,14 @@ private static void appendPythonConfig(Environment env, Configuration pythonConf
 
     public static void main(String[] args) {
         if (args.length < 1) {
-            CliOptionsParser.printHelpClient();
-            return;
+            args = new String[]{MODE_EMBEDDED};
+        } else {
+            String mode = args[0];
+            if (!MODE_EMBEDDED.equals(mode)) {
```

Comment:
We should add the MODE_EMBEDDED only when the first arg starts with `-`  (which are options). 

Otherwise, `sql-client.sh gateway` may changed to `sql-client.sh embedded gateway` which is not expected.